### PR TITLE
Various wording/formatting adjustements

### PR DESCRIPTION
**Pull request by JesseWeinstein**
This includes some additional guidelines, but is not intended to make any substantive changes.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -77,55 +82,59 @@ progressively greater amount of knowledge."
...
-#### Install
+### Install
 **Status:** Required by default, optional for doc modules.
 **Requirements:**
 - Code block illustrating how to install.
 
-Subsections:
-- `Dependencies`. Required if necessary.
+**Subsections:**
+- `Dependencies`. Required if there are dependencies.
```
Comment (RichardLitt) [inline spec.md]: How about `if there are unexpected dependencies which must be independently installed`?

File: spec.md [hunk 2]
```diff
@@ -77,55 +82,59 @@ progressively greater amount of knowledge."
...
-**Status**: Optional.  
+### Extra Sections
+**Status**: Optional.
 **Requirements:**
 - None.
 
-#### API
-**Status:** Optional.  
+**Suggestions:**
+- Does not require its own title.
```
Comment (RichardLitt) [inline spec.md]: Should specify that this encapsulates other possible sections, which must have a title.

File: spec.md [hunk 3]
```diff
@@ -135,8 +144,8 @@ Subsections:
 - Describe caveats.
 - If using an external API generator (like go-doc, js-doc, or so on), point to an external `API.md` file. This can be the only item in the section, if present.
 
-#### Contribute
-**Status**: Required.  
+### Contribute
+**Status**: Required.
```
Comment (RichardLitt) [inline spec.md]: Trailing spaces are needed in Markdown. Please don't remove them.

**Comment by RichardLitt** [comment]
Looks good! Small things to fix.

**Comment by RichardLitt** [comment]
Try running [vmd](https://github.com/yoshuawuyts/vmd) or [marky-markdown](https://www.npmjs.com/package/marky-markdown) to preview output.

File: spec.md (hunk 1 above)
Comment (JesseWeinstein) [inline spec.md]: That sounds better, yes.

File: spec.md (hunk 3 above)
Comment (JesseWeinstein) [inline spec.md]: Trailing spaces are REQUIRED by Markdown? That's a ... horrifying ... design decision. But certainly, I'll avoid removing them.

Comment (JesseWeinstein) [inline spec.md]: Ah, I see, it's a "easier" way to write BR elements. http://www.wilfred.me.uk/blog/2012/07/30/why-markdown-is-not-my-favourite-language/

Can I replace them *with* BR elements, for the sake of the sanity of future editors?

**Comment by JesseWeinstein** [comment]
OK, I think I resolved the issues you pointed out. Let me know if not!

File: spec.md [hunk 4]
```diff
@@ -140,8 +140,8 @@ progressively greater amount of knowledge."
 - None.
 
 **Suggestions:**
-- Does not require its own title.
-- Any other sections that are relevant should be added here, after [Usage](#usage) and before [API](#api).
+- Does not require its own title, however the sections included within it must have titles.
+- This should contain any other sections that are relevant, placed after [Usage](#usage) and before [API](#api).
 - Specfically, the [Security](#security) section should be here if it wasn't important enough to be placed above.
```
Comment (RichardLitt) [inline spec.md]: Spelling error?

**Comment by RichardLitt** [comment]
Thanks @JesseWeinstein! Great work on this, I appreciate it a ton.

File: spec.md (hunk 4 above)
Comment (JesseWeinstein) [inline spec.md]: I don't see an error...?

**Pull request merged**